rust/metering/src/core.rs

```diff
@@ -230,17 +290,27 @@ initialize_metering! {
         pub collection_id: String,
         #[serde(flatten)]
         pub action: WriteAction,
+        pub request_received_at: DateTime<Utc>,
```

Comment:
Would it make more sense to either

- have a start() mutator, that internally sets this, and a finish() mutator that ends it
- directly hydrate a `time_elapsed` capability ?